File: itx-java/itx-backend/src/main/java/io/github/chehsunliu/itx/backend/feature/health/HealthRoutes.java
package io.github.chehsunliu.itx.backend.feature.health;

import io.github.chehsunliu.itx.backend.middleware.Envelope;
import io.javalin.Javalin;
import java.util.Map;

public final class HealthRoutes {
  private HealthRoutes() {}

  public static void register(Javalin app) {
    app.get("/api/v1/health", ctx -> Envelope.data(ctx, Map.of("status", "ok")));
  }
}

File: itx-kt/itx-backend/src/main/kotlin/io/github/chehsunliu/itx/backend/feature/health/HealthRoutes.kt
package io.github.chehsunliu.itx.backend.feature.health

import io.github.chehsunliu.itx.backend.middleware.respondDataObject
import io.ktor.server.application.call
import io.ktor.server.routing.Route
import io.ktor.server.routing.get
import io.ktor.server.routing.route
import kotlinx.serialization.json.JsonPrimitive
import kotlinx.serialization.json.buildJsonObject

fun Route.healthRoutes() {
    route("/health") {
        get {
            call.respondDataObject(buildJsonObject { put("status", JsonPrimitive("ok")) })
        }
    }
}
